components/emc2101.rst

```diff
@@ -0,0 +1,177 @@
+          id: speed_sensor
+          name: "Speed Sensor"
+        duty_cycle:
+          id: duty_cycle_sensor
+          name: "Duty Cycle Sensor"
+        update_interval: 5s
+
+
+Configuration variables:
+************************
+
+- **address** (*Optional*, int): Manually specify the I²C address of the sensor. Defaults to ``0x4C``.
+- **inverted** (*Optional*, boolean): Inverts the output so 0 means 100% duty cycle and 1 means 0%.  Defaults to ``false``.
+- **pwn** (*Optional*): Enable PWN output.
+
+  - **resolution** (*Optional*, int): Determines effective resolution and the frequency of the PWM signal. Defaults to ``23``.
+  - **divider** (*Optional*, int): Value for the frequency divider. Defaults to ``1``.
+
+- **dac** (*Optional*): Enable DAC output.
+
+  - **conversion_rate** (*Optional*, string): The number of digital to analog conversions perfomed per second. One of ``1/16``, ``1/8``,
+    ``1/4``, ``1/2``, ``1``, ``2``, ``4``, ``8``, ``16``, ``32``. Defaults to ``16``.
+
+
+The PWN signal frequency is determined by the following formula:
+
+.. math::
+
+    \text{PWM Frequency} = \frac{360KHz}{2 \times resolution \times divider}
+
+And the PWM resolution, the steps in the duty cycle, is calculated by the following formula:
+
+.. math::
+
+    \text{PWM resolution} = \frac{100\%}{2 \times resolution}
+
+So, with the default values the PWM signal will have a frequency of 7.83KHz and a resolution of 2.17%. The maximum
+resolution value is ``31`` which will configure a frequency of 5.81KHz with a resolution of 1.61%.
+
+
+
+EMC2101 Output
+--------------
+
+.. code-block:: yaml
+
+    output:
+      - platform: emc2101
+        id: fan_duty_cycle
+
+
+Configuration variables:
+************************
+
+- **id** (**Required**, :ref:`config-id`): The ID to use for code generation, and for reference by dependent components.
+
+.. note::
+
+    When using the ``emc2101`` output with a :doc:`/components/fan/speed` it is advised to configure the **speed_count**
+    as twice the **resolution** so the number of speeds in the fan matches the number of steps in the fan controller.
+
+
+
+EMC2101 Sensors
+---------------
+
+.. code-block:: yaml
+
+    sensor:
+      - platform: emc2101
+        internal_temperature:
+          id: internal_temperature_sensor
+          name: "Internal Temperature Sensor"
+        speed:
+          id: speed_sensor
+          name: "Speed Sensor"
+        duty_cycle:
+          id: duty_cycle_sensor
+          name: "Duty Cycle Sensor"
+        update_interval: 5s
+
+
+Configuration variables:
+************************
+- **internal_temperature** (*Optional*): The internal temperature sensor.
+
+  - **id** (*Optional*, string): Manually specify the ID for code generation. At least one of **id** and **name** must be specified.
+  - **name** (*Optional*, string): The name for the sensor. At least one of **id** and **name** must be specified.
+  - All other options from :ref:`Sensor <config-sensor>`.
```

Comment:
This is already defined in the linked sensor config for users to read and learn.
```suggestion
- **internal_temperature** (*Optional*): The internal temperature sensor. 
  All options from :ref:`Sensor <config-sensor>`.
```